O365/financials.py

```diff
@@ -0,0 +1,4048 @@
+                return None
+
+        if not entity_id:
+            raise RuntimeError('Provide one of the options')
+
+        # Everything received from cloud must be passed as self._cloud_data_key
+        return constructor(parent=self, **{self._cloud_data_key: data})
+
+    def _get_entities(self, constructor=None, limit=None, query=None, order_by=None, parent=None):
+        if not constructor:
+            raise RuntimeError('Provide one of the options')
+
+        cid = self.__parent_id if self.__parent_id else self.object_id
+
+        if constructor:
+            if parent:
+                url = self.build_url("/{entity}/{id}/{navigationEntity}".format(entity=self._ENTITY, id=self.object_id, navigationEntity=constructor._ENTITY))
+            else:
+                url = self.build_url("/{cid}/{entity}".format(cid=cid, entity=constructor._ENTITY))
+            params = None
+
+        params = {}
+        if limit:
+            params['$top'] = limit
+        if query:
+            params['$filter'] = str(query)
+        if order_by:
+            params['$orderby'] = order_by
+
+        response = self.con.get(url, params=params)
+        if not response:
+            return []
+
+        data = response.json()
+
+        if not id:
+            raise RuntimeError('Provide one of the options')
+
+        # Everything received from cloud must be passed as self._cloud_data_key
+        return [constructor(parent=self, **{self._cloud_data_key: x}) for x in data.get('value', [])]
+
+    def _update(self):
+        if not self.object_id:
+            return False
+        else:
+            if not self._track_changes:
+                return True  # there's nothing to update
+
+            url = self.build_url(("/" + type(self)._ENTITY + "/{id}").format(id=self.object_id))
+
+            method = self.con.patch
+            data = self.to_api_data(restrict_keys=self._track_changes)
+            headers = {"If-Match": self.__etag}
+
+            response = method(url, data=data, headers=headers)
+
+            if not response:
+                return False
+
+            if not response.ok:
+                return False
+
+            self.__modified = self.protocol.timezone.localize(datetime.now())
+
+            return True
+
+    def _delete(self):
+        if not self.object_id:
+            return False
+
+        url = self.build_url(("/" + type(self)._ENTITY + "/{id}").format(id=self.object_id))
+
+        response = self.con.delete(url)
+        if response.status_code != 204:
+            return False
+
+        self.object_id = None
+        return True
+
+    def _create(self):
+        if not self._track_changes:
+            return True  # there's nothing to update
+
+        url = self.build_url("/" + type(self)._ENTITY)
+
+        method = self.con.post
+        data = self.to_api_data(restrict_keys=self._track_changes)
+
+        response = method(url, data=data)
+
+        if not response.status_code == 201:
```

Comment:
This in my experience with Microsoft Apis will sometimes fail. There are plenty of methods in the graph docs that should return 201 on created events, but returns other status codes. At the beginning of this library I was comparing to the exact status code on each method and I changed this to simply `if not reponse:` or if you prefer `if not response.ok` (both are the same).

Reply:
Done. Thanks for the tip.